core/chains/evm/client/errors.go

```diff
@@ -36,6 +36,9 @@ func (s *SendError) CauseStr() string {
 
 const (
 	NonceTooLow = iota
+	// Nethermind specific error. Nethermind throws a NonceGap error when the tx nonce is greater than current_nonce + tx_count_in_mempool, instead of keeping the tx in mempool.
```

Comment:
Actually seems like we already have a bunch of places in this file where "nonce too high" was treated previously as a fatal error.
Just search for the string "nonce too high" in this file.
I see gethFatal, arbitrumFatal, and erigonFatal being added in the last 6 months in different PRs.

Maybe not in this PR, but let us fix those in a follow-up PR.
https://app.shortcut.com/chainlinklabs/story/59049/nonce-too-high-this-error-is-incorrectly-being-treated-as-fatal-in-evm-code

Reply:
Sounds good. @simsonraj is working on the other chains/clients as part of a separate ticket.